vcr/credential/validator.go

```diff
@@ -176,16 +176,19 @@ func (d nutsAuthorizationCredentialValidator) Validate(credential vc.VerifiableC
 	if len(strings.TrimSpace(cs.PurposeOfUse)) == 0 {
 		return failure("'credentialSubject.PurposeOfUse' is nil")
 	}
-	switch cs.LegalBase.ConsentType {
-	case "implied":
-		// no additional requirements
-		break
-	case "explicit":
-		if cs.Subject == nil || len(strings.TrimSpace(*cs.Subject)) == 0 {
-			return failure("'credentialSubject.Subject' is required when consentType is 'explicit'")
+
+	if credential.ContainsContext(NutsV2ContextURI) {
```

Comment:
This will break as soon there's a V3 context that also implies....
Either make it fool proof or create an issue that we have to get rid of these validators.

Reply:
Question for V3 is when, and what. Long as we have these validators, introducing a new JSON-LD context will include altering these validators. I'm all for getting rid of these validators, but there should be something in return to keep people from making mistakes when issuing credentials (these are all too common). I'd prefer pluggable validation rules, like JSON-LD contexts are. Maybe in the form of JSON schemas.